tests/staticfiles_tests/test_checks.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.core import checks
+from django.test import SimpleTestCase, override_settings
+
+
+@override_settings(
+    INSTALLED_APPS=[
+        'django.contrib.auth',
+        'django.contrib.contenttypes',
+        'django.contrib.staticfiles'
+    ],
+    SILENCED_SYSTEM_CHECKS=['fields.W342'],  # ForeignKey(unique=True)
+)
+class SystemChecksTestCase(SimpleTestCase):
+
+    @override_settings(
+        STATICFILES_DIRS="a string"
+    )
+    def test_filesystemfinder_dirs_not_tuple_or_list(self):
+        """
+        Raise system check error if STATICFILES_DIRS is not
+        a tuple or list.
+        """
+        errors = checks.run_checks()
```

Comment:
Probably the check functions should be called directly rather than invoking them through `run_checks()` (otherwise, this runs all registered checks across all installed apps which doesn't provide good isolation) -- see `tests/check_framework`.